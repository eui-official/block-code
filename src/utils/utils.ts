export const Utils = {
  findIndex: (
    index: number,
    length: number,
    maxLength: number,
    maxInputs: number
  ) => {
    if (length === maxLength && index < maxInputs) {
      ++index
    } else if (length === 0 && index > 0) {
      --index
    }

    return index
  },

  createId: (prefix = 'block-code') =>
    `${prefix}-${Math.random().toString(36).substring(2, 9)}`,

  createChunk: (value: string, length: number) =>
    value.match(new RegExp('.{1,' + length + '}', 'g'))
}
