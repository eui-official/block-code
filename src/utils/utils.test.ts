import { describe, expect, test } from 'vitest'
import { Utils } from './utils'

describe('Utils', () => {
  test('findIndex with default', () => {
    const index = Utils.findIndex(1, 1, 1, 6)
    expect(index).toBe(2)
  })

  test('findIndex with max length to be invalid', () => {
    const index = Utils.findIndex(1, 1, 2, 6)
    expect(index).toBe(1)
  })

  test('findIndex with max length to be valid', () => {
    const index = Utils.findIndex(2, 0, 1, 6)
    expect(index).toBe(1)
  })

  test('findIndex with max length is 0 and index is greater than 0', () => {
    const index = Utils.findIndex(1, 2, 2, 6)
    expect(index).toBe(2)
  })

  test('createId', () => {
    const id = Utils.createId()
    expect(id).toContain('block-code-')
  })

  test('createId with custom predix', () => {
    const id = Utils.createId('input')
    expect(id).toContain('input-')
  })

  test('createChunk', () => {
    const chunk = Utils.createChunk('block-code', 1)
    expect(chunk).toMatchSnapshot()
  })

  test('createChunk with length 2', () => {
    const chunk = Utils.createChunk('block-code', 2)
    expect(chunk).toMatchSnapshot()
  })

  test('createChunk with length 2 but value has spaces', () => {
    const chunk = Utils.createChunk('r eact - code', 2)
    expect(chunk).toMatchSnapshot()
  })
})
