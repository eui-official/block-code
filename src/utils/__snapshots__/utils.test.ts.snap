// Vitest Snapshot v1

exports[`Utils > createChunk 1`] = `
[
  "b",
  "l",
  "o",
  "c",
  "k",
  "-",
  "c",
  "o",
  "d",
  "e",
]
`;

exports[`Utils > createChunk with length 2 1`] = `
[
  "bl",
  "oc",
  "k-",
  "co",
  "de",
]
`;

exports[`Utils > createChunk with length 2 but value has spaces 1`] = `
[
  "r ",
  "ea",
  "ct",
  " -",
  " c",
  "od",
  "e",
]
`;
